### PR TITLE
Update info on PreBuildEvent / PostBuildEvent

### LINKED ISSUE #15248: Mention that PreBuiltEvent and PostBuildEvent shouldn't be used (since common macro properties don't work there)
Docs about the new project file format should mention that macro properties like $(ProjectDir) [don't work](https://github.com/dotnet/sdk/issues/677#issuecomment-280164111) in PreBuildEvent and PostBuildEvent properties when using SDK-style projects. Instead, users should create new targets (using the 'exec' command) that run before or after the PreBuildEvent and PostBuildEvent targets.

Visual Studio has [already been updated](https://github.com/dotnet/project-system/issues/1569#issuecomment-360951567) so that any pre/post build commands entered through the project properties UI will create new targets instead of using PreBuildEvent/PostBuildEvent directly. We should add a note to the docs, though, explaining this so that users migrating old csproj file to the new SDK-style format know that they will need to change PreBuildEvent and PostBuildEvent usage.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb4428fc-215c-78e1-5ff0-fe73f7a08260
* Version Independent ID: 6741b1de-38a4-fbb1-d79d-acc47d549a2d
* Content: [Additions to the csproj format for .NET Core - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj)
* Content Source: [docs/core/tools/csproj.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/csproj.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by ghogen**
## Summary

Add a new section "Build events" that shows how the old vs. new project file format encodes pre- and post-build events. This is important for those migrating projects or manually editing project files.

Fixes #15248

### CODE REVIEW

File: docs/core/tools/csproj.md [hunk 1]
```diff
@@ -223,6 +223,24 @@ The following example specifies the fallbacks only for the `netcoreapp2.1` targe
...
+
+The way that pre-build and post-build events are specified in the project file has changed. The properties PreBuildEvent and PostBuildEvent are not recommended in the SDK-style project format, because macros such as $(ProjectDir) are not resolved. For example, the following code is no longer supported:
+
+```xml
+<PropertyGroup>
+    <PreBuildEvent>"$(ProjectDir)PreBuildEvent.bat" "$(ProjectDir)..\" "$(ProjectDir)" "$(TargetDir)" />
+</PropertyGroup>
+```
+
+In SDK-style projects, use an MSBuild target named `PreBuild` or `PostBuild`. For the preceding example, use the following code:
```
Comment (Thraka) [inline docs/core/tools/csproj.md]: Aren't the names `PreBuild` and `PostBuild` arbitrary? Really the requirement is to create a new target that calls `BeforeTargets="PreBuildEvent"` and `AfterTargets="PostBuildEvent"`

Yes?

Comment (ghogen) [reply docs/core/tools/csproj.md]: You're probably right, let me confirm and update as necessary.

Comment (ghogen) [reply docs/core/tools/csproj.md]: What I found is that if you use different names for the targets, they do run as expected.  However, the VS IDE does not find and detect them and show them in the Build Events tab of the Project Designer, where they can be edited.  I could add a note that explains that, maybe.

Comment (Thraka) [reply docs/core/tools/csproj.md]: Interesting! I think that it's OK as it is, but maybe just add another sentence explaining that each should be setting the Before/After targets property.

**Review by Thraka — APPROVED** [review]
LGTM Thanks!

**Comment by ghogen** [comment]
@thraka I think this is ready now.

**Pull request merged**